File: PartnersAPI/Model/Partner.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PartnersAPI.Models
{
	public class Partner
	{
		[Key]
		public int PartnerId { get; set; }
		[Required]
		public string BusinessName { get; set; } = string.Empty;
		[Required]
		public string Address { get; set; } = string.Empty;
		[Required]
		public string PhoneNumber { get; set; } = string.Empty;
		[Required]
		public string ContactPerson { get; set; } = string.Empty;
		[Required]
		public string Email { get; set; } = string.Empty;

		[NotMapped]
		public virtual Credentials Credentials { get; set; } = default!;
	}
}

File: partners_app/lib/models/partner.dart
class Partner {
  int id;
  String businessName;
  String address;
  String phoneNumber;
  String contactPerson;
  String email;
  int reach;

  Partner(
      {required this.id,
      required this.businessName,
      required this.address,
      required this.phoneNumber,
      required this.contactPerson,
      required this.email,
      required this.reach});

  factory Partner.fromJson(Map<String, dynamic> json) {
    return Partner(
      id: json['partner_id'],
      businessName: json['business_name'],
      address: json['address'],
      phoneNumber: json['phone_number'],
      contactPerson: json['contact_person'],
      email: json['email'],
      reach: json['reach'],
    );
  }
}
